src/fileFrom/fileFrom.ts

```diff
@@ -1,63 +1,105 @@
-import { UploadLifecycle } from '../lifecycle/UploadLifecycle'
-import { FileUploadLifecycle } from '../lifecycle/FileUploadLifecycle'
-import { FileFromObject } from './FileFromObject'
-import { FileFromUploaded } from './FileFromUploaded'
-import { FileFromUrl } from './FileFromUrl'
+import fromObject from './fromObject'
+import CancelController from '../CancelController'
+import defaultSettings from '../defaultSettings'
 
 /* Types */
-import { FileData, SettingsInterface, UploadcareFileInterface } from '../types'
-import { Url } from '../api/fromUrl'
-import { Uuid } from '../api/types'
-import {
-  FileUploadLifecycleInterface,
-  LifecycleHooksInterface,
-  UploadInterface
-} from '../lifecycle/types'
+import { FileData, UploadcareFileInterface } from '../types'
+import { Url, Uuid } from '../api/types'
 import { isFileData, isUrl, isUuid } from './types'
-import { Upload } from '../lifecycle/Upload'
 
 /**
  * Uploads file from provided data.
- *
- * @param {FileData | Url | Uuid} data
- * @param {SettingsInterface} settings
- * @param {LifecycleHooksInterface<UploadcareFileInterface>} hooks
- * @throws TypeError
- * @returns {UploadInterface<UploadcareFileInterface>}
+ * @param data
+ * @param options
+ * @param [options.publicKey]
+ * @param [options.fileName]
+ * @param [options.baseURL]
+ * @param [options.secureSignature]
+ * @param [options.secureExpire]
+ * @param [options.store]
+ * @param [options.cancel]
+ * @param [options.onProgress]
+ * @param [options.source]
+ * @param [options.integration]
+ * @param [options.retryThrottledRequestMaxTimes]
  */
 export default function fileFrom(
   data: FileData | Url | Uuid,
-  settings: SettingsInterface = {},
-  hooks?: LifecycleHooksInterface<UploadcareFileInterface>
-): UploadInterface<UploadcareFileInterface> {
-  const lifecycle = new UploadLifecycle<UploadcareFileInterface>(hooks)
-  const fileUploadLifecycle = new FileUploadLifecycle(lifecycle)
+  {
+    publicKey,
 
+    fileName = defaultSettings.fileName,
+    baseURL = defaultSettings.baseURL,
+    secureSignature,
+    secureExpire,
+    store,
+
+    cancel,
+    onProgress,
+
+    source,
+    integration,
+
+    retryThrottledRequestMaxTimes,
+
+    baseCDN = defaultSettings.baseCDN,
+    defaultEffects
```

Comment:
we receive `defaultEffects` from fileInfo
see https://uploadcare.com/docs/api_reference/upload/groups/#example

Reply:
Hmm, but in https://uploadcare.com/api-refs/upload-api/#operation/fileUploadInfo there is no any word about this. @andrewshkovskii , can you help?

Reply:
because this works only in groups
https://uploadcare.com/api-refs/upload-api/#operation/filesGroupInfo